blockchains/eth/eth_worker.js

```diff
@@ -0,0 +1,254 @@
+        transactionHash: trace["transactionHash"],
+        transactionPosition: web3.utils.hexToNumber(trace["transactionPosition"]),
+        type: trace["type"]
+      }
+    }
+
+    if (trace["type"] != "call") {
+      logger.warn("Unknown trace type: " + JSON.stringify(trace))
+    }
+
+    return {
+      from: trace["action"]["from"],
+      to: trace["action"]["to"],
+      value: parseFloat(web3.utils.hexToNumberString(trace["action"]["value"])),
+      valueExactBase36: web3.utils.toBN(trace["action"]["value"]).toString(36),
+      blockNumber: trace["blockNumber"],
+      timestamp: timestamp,
+      transactionHash: trace["transactionHash"],
+      transactionPosition: web3.utils.hexToNumber(trace["transactionPosition"]),
+      type: trace["type"]
+    }
+  }
+
+  async fetchBlocks(fromBlock, toBlock) {
+    const blockRequests = []
+    for (let i = fromBlock; i <= toBlock; i++) {
+      blockRequests.push(
+        this.parityClient.request(
+          'eth_getBlockByNumber',
+          [this.web3.utils.numberToHex(i), true],
+          undefined,
+          false
+        )
+      )
+    }
+
+    const responses = await this.parityClient.request(blockRequests);
+    const result = new Map()
+    responses.forEach((response, index) => result.set(fromBlock + index, response.result))
+    return result
+  }
+
+  async fetchReceipts(blocks) {
+    const responses = []
+
+    blocks.forEach((block, blockNumber) => {
+      const req = this.parityClient.request('parity_getBlockReceipts', [this.web3.utils.numberToHex(blockNumber)], undefined, false)
+      responses.push(this.parityClient.request([req]))
+    })
+
+    const finishedRequests = await Promise.all(responses)
+    const result = {}
+
+    finishedRequests.forEach((blockResponses) => {
+      if (!blockResponses) return
+
+      blockResponses.forEach((blockResponse) => {
+        if (blockResponse.result) {
+          blockResponse.result.forEach((receipt) => {
+            result[receipt.transactionHash] = receipt
+          })
+        }
+      })
+    })
+
+    return result
+  }
+
+  async getPastEvents(fromBlock, toBlock) {
+    const web3 = this.web3
+    logger.info(`Fetching traces for blocks ${fromBlock}:${toBlock}`)
+
+    const [traces, blocks] = await Promise.all([
+      this.fetchEthInternalTrx(fromBlock, toBlock),
+      this.fetchBlocks(fromBlock, toBlock)
+    ])
+
+    let result = []
+
+    for (let i = 0; i < traces.length; i++) {
+      result.push(decodeTransferTrace(traces[i], blocks))
+    }
+
+    logger.log(`Fetching receipts of ${fromBlock}:${toBlock}`)
+    const receipts = await fetchReceipts(blocks)
+
+    blocks.forEach((block) => {
+      block.transactions.forEach((transaction) =>
+        result.push({
+          from: transaction.from,
+          to: block.miner,
+          value: parseFloat(web3.utils.hexToNumberString(transaction.gasPrice)) * parseFloat(web3.utils.hexToNumberString(receipts[transaction.hash].gasUsed)),
```

Comment:
Hard to read, better extract to a function and give meaningful name.

Reply:
Done.